src/event_pusher/mod_event_pusher_push.erl

```diff
@@ -165,17 +168,20 @@ maybe_enable_node(#jid{lserver = Host} = BareFrom, BarePubSubJID, Node, FormFiel
 %% Router callbacks
 %%--------------------------------------------------------------------
 
--spec handle_publish_response(BareRecipient :: jid:jid(), PubsubJID :: jid:jid(),
+-spec handle_publish_response(Recipient :: jid:jid(), PubsubJID :: jid:jid(),
                               Node :: pubsub_node(), Result :: timeout | jlib:iq()) -> ok.
-handle_publish_response(_BareRecipient, _PubsubJID, _Node, timeout) ->
+handle_publish_response(_Recipient, _PubsubJID, _Node, timeout) ->
     ok;
-handle_publish_response(_BareRecipient, _PubsubJID, _Node, #iq{type = result}) ->
+handle_publish_response(_Recipient, _PubsubJID, _Node, #iq{type = result}) ->
     ok;
-handle_publish_response(BareRecipient, PubsubJID, Node, #iq{type = error, sub_el = Els}) ->
+handle_publish_response(Recipient, PubsubJID, Node, #iq{type = error, sub_el = Els}) ->
     [Error | _ ] = [Err || #xmlel{name = <<"error">>} = Err <- Els],
     case exml_query:attr(Error, <<"type">>) of
         <<"cancel">> ->
             %% We disable the push node in case the error type is cancel
+            ejabberd_sm:remove_info(Recipient#jid.luser, Recipient#jid.lserver, Recipient#jid.lresource,
```

Comment:
is it really correct? cause we don't really know if this node is associated with this session or some another.

i would say that we should not care about this case at all.

Reply:
we can try to remove the node from all the sessions associated with the user, if you really want to do it. but technically removing of the node from the session info is not critical for us at all.

Reply:
I changed it to removing the push node from sessions it belongs to.